src/main/scala/in/ashwanthkumar/suuchi/router/ReplicationRouter.scala

```diff
@@ -66,36 +69,90 @@ abstract class ReplicationRouter(nrReplicas: Int, self: MemberAddress) extends S
     clientResponse
   }
 
+  def forwardAsync[RespT, ReqT](methodDescriptor: MethodDescriptor[ReqT, RespT], headers: Metadata,
+                                incomingRequest: ReqT,
+                                destination: MemberAddress)(implicit executor: Executor): ListenableFuture[RespT] = {
+    forwarded = true
```

Comment:
@brewkode I just realised we shouldn't model this like this. The ServerIntercept is not threadsafe la. Meaning there's only 1 version of the ServerIntercept object in a service lifetime. We can only hold state inside the Listener that we pass out. I guess we might have to model the whole thing like that. 

I know this is not related to this change, but something that just stuck me.